src/util/heap/layout/map64.rs

```diff
@@ -28,15 +28,20 @@ pub struct Map64 {
     cumulative_committed_pages: AtomicUsize,
 }
 
-impl Map for Map64 {
-    type FreeList = RawMemoryFreeList;
-
-    fn new() -> Self {
+impl Map64 {
+    pub fn new() -> Self {
         let mut high_water = vec![Address::ZERO; MAX_SPACES];
         let mut base_address = vec![Address::ZERO; MAX_SPACES];
 
+        assert!(
```

Comment:
If `Map64` does not work on 32 bits, we can guard the entire module with `#[cfg(target_pointer_width = "64")]`.